sourcegraph/cody
# NLS search for OneBox

closes: https://linear.app/sourcegraph/issue/CODY-4453/integerate-nls-api

closes: https://linear.app/sourcegraph/issue/SRCH-1414/integrate-repo-file-dir-mentions-with-search-filters

Currently for search intent, simply the context items fetched using the `getCodyContext` API (which we use to pass to the LLM) were shown as search results. 

This PR uses the graphql search api with `patternType=NLS` to fetch the search results for the query and display them in the assistant response. 

This PR also converts the file, repo & directory mentions to search filters and ignores the rest mentions for search intent.

All UX improvements will be done as separate PRs. 

The 2 blue boxes for "Ask the LLM" and "query" will be only showed if `cody-experimental-one-box-debug` feature flags are enabled.

![CleanShot 2024-12-05 at 19 09 46@2x](https://github.com/user-attachments/assets/8058c8a7-395a-47f8-bec0-208551c59373)


## Test plan

- Enable FF `cody-experimental-one-box`
- Make a search query and check the results.
- Do a follow up chat to make sure it works. 
- Test manually triggered search intent. 

## Changelog
